lib/parsers/mp3_parser.rb

```diff
@@ -20,7 +20,7 @@ class InvalidDeepFetch < KeyError
 
   # We limit the number of MPEG frames we scan
   # to obtain our duration estimation
-  MAX_FRAMES_TO_SCAN = 128
+  MAX_FRAMES_TO_SCAN = 500
```

Comment:
Just curious, based on what did you get to 500?

Reply:
In this file that I used to test, the first frame was found only after the 367th read, so I changed to 500 to give a space for other files which can have that afterwards

Maybe that's something we can improve after, but I'm not sure how yet

wdyt?

Reply:
Is there a standard/guideline on when to expect that first frame within mp3s? 😀  Feels like shooting for the stars here, and it would be nice to make it work for everything, if we can.

I mostly asked because 128 is a _round_ number and 500 isn't 😁

Reply:
The doc is here http://www.mpgedit.org/mpgedit/mpeg_format/mpeghdr.htm

>The frame header is constituted by the very first four bytes (32bits) in a frame. The first eleven bits (or first twelve bits, see below about frame sync) of a frame header are always set and they are called "frame sync". Therefore, you can search through the file for the first occurence of frame sync (meaning that you have to find a byte with a value of 255, and followed by a byte with its three (or four) most significant bits set)

From what I understand, the frames don't have a fixed size and we need to find the next "frame sync" to find them.

But, the implementation in format_parser, has a "protection" against huge files, and because of that, it reads every 4 bytes and it has a limit, which was 128 and I increased to 500.

So, I don't know how to define this number more precisely.

Do you have a suggestion?

Reply:
Thanks for linking the spec. Let's go with this and see how it behaves in practice 👍

Reply:
great!!